### DISCUSSION TITLE
Scripted or dynamic choices

### DISCUSSION THREAD

**Discussion by sgarfinkel** [copier-org/copier#1025] [Q&A]
I would like to retrieve choices for allowing users to select a field from and LDAP server. Is there anyway to get choices from a script?

I did a little testing and I can see that there’s no way to pass a jinja template as the top-level choices, since a string is an invalid element for choices.

The only hacky way I can think of doing this would be to emit a yaml file with the question that gets included in the base `copier.yaml` but this would require some way to run a task _before_ prompting, which as far as I know is not possible.

**Comment by yajo** [comment]
You'd need some jinja extension to do that.
